paasta_tools/cli/cmds/start_stop_restart.py

```diff
@@ -167,91 +174,77 @@ def print_chronos_message(desired_state):
-    # provided in the -i argument is deployed there.
-    # if there are no instances defined in the args, then assert that the service
-    # is deployed to that cluster.
-    # If args.clusters is falsey, then default to *all* clusters that a service is deployed to,
-    # and we figure out which ones are needed for each service later.
-    if instances:
-        instance_clusters = [list_clusters(service, soa_dir, instance) for instance in instances]
-        valid_clusters = sorted(list({cluster for cluster_list in instance_clusters for cluster in cluster_list}))
-    else:
-        valid_clusters = list_clusters(service, soa_dir)
-
-    if pargs.cluster_whitelist:
-        clusters = pargs.cluster_whitelist
-        invalid_clusters = [cluster for cluster in clusters if cluster not in valid_clusters]
-        if invalid_clusters:
-            paasta_print(
-                "Invalid cluster name(s) specified: %s." % " ".join(invalid_clusters),
-                "Valid options: %s" % " ".join(valid_clusters),
-            )
-            return 1
-    else:
-        clusters = valid_clusters
-
-    try:
-        remote_refs = remote_git.list_remote_refs(utils.get_git_url(service, soa_dir))
-    except remote_git.LSRemoteException as e:
-        msg = (
-            "Error talking to the git server: %s\n"
-            "This PaaSTA command requires access to the git server to operate.\n"
-            "The git server may be down or not reachable from here.\n"
-            "Try again from somewhere where the git server can be reached, "
-            "like your developer environment."
-        ) % str(e)
-        paasta_print(msg)
-        return 1
+    affected_services = {s for service_list in pargs.values() for s in service_list.keys()}
+    if len(affected_services) > 1:
+        paasta_print(PaastaColors.red("Warning: trying to start/stop/restart multiple services:"))
 
-    invalid_deploy_groups = []
-    marathon_message_printed, chronos_message_printed = False, False
-    for cluster in clusters:
-        # If they haven't specified what instances to act on, do it for all of them.
-        # If they have specified what instances, only iterate over them if they're
-        # actually within this cluster.
-        if instances is None:
-            cluster_instances = list_all_instances_for_service(service, clusters=[cluster], soa_dir=soa_dir)
-            paasta_print("no instances specified; restarting all instances for service")
+        for cluster, services_instances in pargs.items():
+            paasta_print("Cluster %s:" % cluster)
+            for service, instances in services_instances.items():
+                paasta_print("    Service %s:" % service)
+                paasta_print("        Instances %s" % ",".join(instances))
+
+        if args.allow_multiple_services:
+            paasta_print(PaastaColors.red("    --allow-multiple-services specified, continuing anyways"))
```

Comment:
I don't think that having both a flag and a the choice interaction is great. I'd just use the choice interaction instead, I think.